protocol/communities/manager.go

```diff
@@ -348,19 +360,93 @@ func (m *Manager) Start() error {
 }
 
 func (m *Manager) runENSVerificationLoop() {
+	ticker := time.NewTicker(500 * time.Millisecond)
+	defer ticker.Stop()
 	go func() {
 		for {
 			select {
 			case <-m.quit:
-				m.logger.Debug("quitting ens verification loop")
+				m.logger.Debug("quitting owner verification loop")
 				return
 			case records, more := <-m.ensSubscription:
 				if !more {
 					m.logger.Debug("no more ens records, quitting")
 					return
 				}
 				m.logger.Info("received records", zap.Any("records", records))
+			}
+		}
+	}()
+}
+
+func (m *Manager) runOwnerVerificationLoop() {
+	m.logger.Info("starting owner verification loop")
+	go func() {
+		for {
+			select {
+			case <-m.quit:
+				m.logger.Debug("quitting ens verification loop")
+				return
+			case <-time.After(validateInterval):
+				// If ownerverifier is nil, we skip, this is useful for testing
+				if m.ownerVerifier == nil {
+					continue
+				}
+
+				communitiesToValidate, err := m.persistence.GetCommunitiesToValidate()
+
+				if err != nil {
+					m.logger.Error("failed to fetch communities to validate", zap.Error(err))
+					continue
+				}
+				for id, communities := range communitiesToValidate {
+					m.logger.Info("validating communities", zap.String("id", id), zap.Int("count", len(communities)))
+
+					for _, communityToValidate := range communities {
+						signer, description, err := m.unwrapCommunityDescriptionMessage(communityToValidate.payload)
+						if err != nil {
+							m.logger.Error("failed to unwrap community", zap.Error(err))
+							continue
+						}
+
+						chainID := CommunityDescriptionTokenOwnerChainID(description)
+						if chainID == 0 {
+							// This should not happen
+							m.logger.Error("chain id is 0, ignoring")
+						}
+
+						m.logger.Info("validating community", zap.String("id", types.EncodeHex(communityToValidate.id)), zap.String("signer", common.PubkeyToHex(signer)))
+
+						ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
+						defer cancel()
 
+						owner, err := m.ownerVerifier.SafeGetSignerPubKey(ctx, chainID, id)
+						if err != nil {
+							m.logger.Error("failed to get owner", zap.Error(err))
+							continue
+						}
+
+						ownerPK, err := common.HexToPubkey(owner)
+						if err != nil {
+							m.logger.Error("failed to convert pk string to ecdsa", zap.Error(err))
+							continue
+						}
+
+						response, err := m.HandleCommunityDescriptionMessage(signer, description, communityToValidate.payload, ownerPK)
+						if err != nil {
+							m.logger.Error("failed to handle community", zap.Error(err))
+							continue
+						}
+						if response != nil {
+							m.publish(&Subscription{Community: response.Community, TokenCommunityValidated: response})
```

Comment:
Just to be sure, this `publish` is local only right? It doesn't publish to peers?

Reply:
Correct, it publishes only to `messenger` that can handle the message then and signal to the client